File: 0693-binary-number-with-alternating-bits/0693-binary-number-with-alternating-bits.py
class Solution:
    def hasAlternatingBits(self, n: int) -> bool:
        binary = self.getBinary(n)
        
        for i in range(1, len(binary)):
            if(binary[i]=="0"):
                if(binary[i-1]!="1"):
                    return False
            else:
                if(binary[i-1]!="0"):
                    return False
        
        return True
    
    def getBinary(self, n):
        res = ""
        while(n>1):
            rem = n%2
            res = str(rem)+res
            n= n//2
        
        res = "1"+res
        return res

File: 0693-binary-number-with-alternating-bits/0693-binary-number-with-alternating-bits.java
class Solution {
    public boolean hasAlternatingBits(int n) {
        String binary = getBinary(n);

        for(int i = 1; i<binary.length(); i++){
            if(binary.charAt(i)=='0'){
                if(binary.charAt(i-1)!='1'){
                    return false;
                }
            }
            else{
                if(binary.charAt(i-1)!='0'){
                    return false;
                }
            }
        }
        
        return true;
    }

    public String getBinary(int n){
        String res = "";

        while(n>1){
            int rem = n%2;
            res = rem+res;
            n = n/2;
        }
        
        res = "1"+res;

        return res;
    }
}